src/storage/src/render/sources.rs

```diff
@@ -436,33 +436,13 @@ where
         error_collections.push(errors.as_collection());
     };
 
-    // Flatten the error collections.
-    let mut err_collection = match error_collections.len() {
+    // Fu the error collections.
+    let err_collection = match error_collections.len() {
         0 => Collection::empty(scope),
         1 => error_collections.pop().unwrap(),
         _ => collection::concatenate(scope, error_collections),
     };
 
-    // Apply `as_of` to each timestamp.
```

Comment:
🙌 Before I got down to this change I was wondering whether we should have those old uses of the `as_of` anymore. Because it doesn't make sense to advance it here, before writing it to persist.